apkid/rules/dex/obfuscators.yara

```diff
@@ -241,3 +242,23 @@ rule aamo_str_enc : obfuscator
   condition:
     1 of ($opcodes*) and all of ($a, $b)
 }
+
+rule appsuit : obfuscator
+{
+    meta:
+        description = "AppSuit"
+        url         = "http://www.stealien.com/appsuit.html"
+        example     = "b99bafbbd5288ac93647d22f1c5b1863c96f581ae7a19fdc0e84bff4c2141328"
+        author      = "Eduardo Novella"
+
+    strings:
+        $class1      = "Lcom/stealien/const"
+        $class2      = "Lcom/stealien/appsuit"
+        $class3      = "La_lock"
+        $native_lib  = "AppSuit"
+        $packer      = "AppSuitDexLoader"
+
+    condition:
+        any of them and not is_apk
```

Comment:
This seems likely to FP, mostly `AppSuit`. Could you do `2` or `3 of them` instead?

Reply:
Nope, `2` or `3` doesn't give good results.  I can try to improve the `AppSuit` string if needed. But this rule has been using for months without FP at the moment at Koodous.

Reply:
That's good to know it hasn't FPed yet. Though, if someone runs this rule set on a corpus of 120 million apks, I don't know if I'd be confident "AppSuit" wouldn't show up at least once either alone or part of another string. I've seen some less likely stuff FP, and I doubt we're really pushing any technical barriers by going for more specific rules.

Reply:
Hi @CalebFenton,

Take a peek at:

```
[22:50 edu@de11 2] >  for p in `ls **/*`; do strings -f $p 2>/dev/null|egrep -i "(appsuit|stealien)"|sort; done
08f0214f055c0ff2697e00b961a2791e74515b64823998a94383f0f2cfb2c13a.apk.unpack/classes.dex: AppSuit
08f0214f055c0ff2697e00b961a2791e74515b64823998a94383f0f2cfb2c13a.apk.unpack/classes.dex: Lcom/stealien/const;
147e3ac633852ba69ace66c6b423187cb1a917bf178703a1d617422fdc8f39c7.apk.unpack/classes.dex: AppSuit
147e3ac633852ba69ace66c6b423187cb1a917bf178703a1d617422fdc8f39c7.apk.unpack/classes.dex: 0Lcom/kbstar/kbbank/base/util/kbaPPsUITmANAGER$2;
147e3ac633852ba69ace66c6b423187cb1a917bf178703a1d617422fdc8f39c7.apk.unpack/classes.dex: 1Lcom/kbstar/kbbank/base/util/kb/AppSuitManager$1;
147e3ac633852ba69ace66c6b423187cb1a917bf178703a1d617422fdc8f39c7.apk.unpack/classes.dex: 1Lcom/kbstar/kbbank/base/util/kb/AppSuitManager$2;
147e3ac633852ba69ace66c6b423187cb1a917bf178703a1d617422fdc8f39c7.apk.unpack/classes.dex: 1Lcom/kbstar/kbbank/base/util/kb/AppSuitManager$3;
147e3ac633852ba69ace66c6b423187cb1a917bf178703a1d617422fdc8f39c7.apk.unpack/classes.dex: 1Lcom/kbstar/kbbank/base/util/kb/AppSuitManager$4;
147e3ac633852ba69ace66c6b423187cb1a917bf178703a1d617422fdc8f39c7.apk.unpack/classes.dex: 1Lcom/kbstar/kbbank/base/util/kb/AppSuitManager$5;
147e3ac633852ba69ace66c6b423187cb1a917bf178703a1d617422fdc8f39c7.apk.unpack/classes.dex: 1Lcom/kbstar/kbbank/base/util/kb/AppSuitManager$6;
147e3ac633852ba69ace66c6b423187cb1a917bf178703a1d617422fdc8f39c7.apk.unpack/classes.dex: checkAppSuit
147e3ac633852ba69ace66c6b423187cb1a917bf178703a1d617422fdc8f39c7.apk.unpack/classes.dex: DLcom/kbstar/kbbank/base/util/kb/AppSuitManager$RequestTokenListener;
147e3ac633852ba69ace66c6b423187cb1a917bf178703a1d617422fdc8f39c7.apk.unpack/classes.dex: FLcom/kbstar/kbbank/base/util/kb/AppSuitManager$RequestConfirmListener;
147e3ac633852ba69ace66c6b423187cb1a917bf178703a1d617422fdc8f39c7.apk.unpack/classes.dex: getProcessAppSuitChallenge
147e3ac633852ba69ace66c6b423187cb1a917bf178703a1d617422fdc8f39c7.apk.unpack/classes.dex: Lcom/stealien/const;
147e3ac633852ba69ace66c6b423187cb1a917bf178703a1d617422fdc8f39c7.apk.unpack/classes.dex: /Lcom/kbstar/kbbank/base/util/kb/AppSuitManager;
147e3ac633852ba69ace66c6b423187cb1a917bf178703a1d617422fdc8f39c7.apk.unpack/classes.dex: runRequestAppSuitChallenge
147e3ac633852ba69ace66c6b423187cb1a917bf178703a1d617422fdc8f39c7.apk.unpack/classes.dex: showAppSuitErrorDialog
2b85dedf0cff405bcabd9a43945aca5f1ee94864ef54f500e1d576a1f0792024.apk.unpack/classes.dex: AppSuit
2b85dedf0cff405bcabd9a43945aca5f1ee94864ef54f500e1d576a1f0792024.apk.unpack/classes.dex: Lcom/stealien/const;
8dc42cc950617ff51d0409a05809d20ca4c375f05c3fa2324b249e1306758a94.apk.unpack/classes2.dex: processAppsuitConfirm
8dc42cc950617ff51d0409a05809d20ca4c375f05c3fa2324b249e1306758a94.apk.unpack/classes2.dex: showAppSuitErrorDialog
8dc42cc950617ff51d0409a05809d20ca4c375f05c3fa2324b249e1306758a94.apk.unpack/classes.dex: AppSuit
8dc42cc950617ff51d0409a05809d20ca4c375f05c3fa2324b249e1306758a94.apk.unpack/classes.dex: Lcom/stealien/const;
a068a94203f8cca6562dbb9e30802f6cec611b8208458bc2905d9baa8cfdf97e.apk.unpack/classes.dex: AppSuit
a068a94203f8cca6562dbb9e30802f6cec611b8208458bc2905d9baa8cfdf97e.apk.unpack/classes.dex: Lcom/stealien/const;
acc9e29a29bdf73b939a0fc1b3f9470a1ee69b668222887878d0ebd56551e909.apk.unpack/classes.dex: KLcom/stealien/
acc9e29a29bdf73b939a0fc1b3f9470a1ee69b668222887878d0ebd56551e909.apk.unpack/classes.dex: SLcom/stealien/appsuit/
acc9e29a29bdf73b939a0fc1b3f9470a1ee69b668222887878d0ebd56551e909.apk.unpack/classes.dex: SLcom/stealien/appsuit/
acc9e29a29bdf73b939a0fc1b3f9470a1ee69b668222887878d0ebd56551e909.apk.unpack/classes.dex: SLcom/stealien/appsuit/
acc9e29a29bdf73b939a0fc1b3f9470a1ee69b668222887878d0ebd56551e909.apk.unpack/classes.dex: SLcom/stealien/appsuit/
acc9e29a29bdf73b939a0fc1b3f9470a1ee69b668222887878d0ebd56551e909.apk.unpack/classes.dex: SLcom/stealien/appsuit/
acc9e29a29bdf73b939a0fc1b3f9470a1ee69b668222887878d0ebd56551e909.apk.unpack/classes.dex: SLcom/stealien/appsuit/
b99bafbbd5288ac93647d22f1c5b1863c96f581ae7a19fdc0e84bff4c2141328.apk.unpack/classes2.dex: APPSUIT
b99bafbbd5288ac93647d22f1c5b1863c96f581ae7a19fdc0e84bff4c2141328.apk.unpack/classes2.dex: stealien
b99bafbbd5288ac93647d22f1c5b1863c96f581ae7a19fdc0e84bff4c2141328.apk.unpack/classes2.dex: stealien_aes_key
b99bafbbd5288ac93647d22f1c5b1863c96f581ae7a19fdc0e84bff4c2141328.apk.unpack/classes2.dex: 1Lcom/stealien/product/keypadsuit/view/KeyPad$2$1;
b99bafbbd5288ac93647d22f1c5b1863c96f581ae7a19fdc0e84bff4c2141328.apk.unpack/classes2.dex: 4Lcom/stealien/product/keypadsuit/encryption/SelfAES;
b99bafbbd5288ac93647d22f1c5b1863c96f581ae7a19fdc0e84bff4c2141328.apk.unpack/classes2.dex: 5Lcom/stealien/product/keypadsuit/button/KeyPadButton;
b99bafbbd5288ac93647d22f1c5b1863c96f581ae7a19fdc0e84bff4c2141328.apk.unpack/classes2.dex: 6[Lcom/stealien/product/keypadsuit/button/KeyPadButton;
b99bafbbd5288ac93647d22f1c5b1863c96f581ae7a19fdc0e84bff4c2141328.apk.unpack/classes2.dex: 8Lcom/stealien/product/keypadsuit/button/DelKeyPadButton;
b99bafbbd5288ac93647d22f1c5b1863c96f581ae7a19fdc0e84bff4c2141328.apk.unpack/classes2.dex: 8Lcom/stealien/product/keypadsuit/generator/PinGenerator;
b99bafbbd5288ac93647d22f1c5b1863c96f581ae7a19fdc0e84bff4c2141328.apk.unpack/classes2.dex: 9Lcom/stealien/product/keypadsuit/button/NullKeyPadButton;
b99bafbbd5288ac93647d22f1c5b1863c96f581ae7a19fdc0e84bff4c2141328.apk.unpack/classes2.dex: BLcom/stealien/product/keypadsuit/generator/ServerSyncPinGenerator;
b99bafbbd5288ac93647d22f1c5b1863c96f581ae7a19fdc0e84bff4c2141328.apk.unpack/classes2.dex: com.stealien.product.keypadsuit
b99bafbbd5288ac93647d22f1c5b1863c96f581ae7a19fdc0e84bff4c2141328.apk.unpack/classes2.dex: DLcom/stealien/product/keypadsuit/encryption/DefaultServerEncryption;
b99bafbbd5288ac93647d22f1c5b1863c96f581ae7a19fdc0e84bff4c2141328.apk.unpack/classes2.dex: DLcom/stealien/product/keypadsuit/keypad/listener/InputEventListener;
b99bafbbd5288ac93647d22f1c5b1863c96f581ae7a19fdc0e84bff4c2141328.apk.unpack/classes2.dex: #Lcom/stealien/product/keypadsuit/a;
b99bafbbd5288ac93647d22f1c5b1863c96f581ae7a19fdc0e84bff4c2141328.apk.unpack/classes2.dex: %Lcom/stealien/product/keypadsuit/a/a;
b99bafbbd5288ac93647d22f1c5b1863c96f581ae7a19fdc0e84bff4c2141328.apk.unpack/classes2.dex: %Lcom/stealien/product/keypadsuit/a/b;
b99bafbbd5288ac93647d22f1c5b1863c96f581ae7a19fdc0e84bff4c2141328.apk.unpack/classes2.dex: 'Lcom/stealien/product/keypadsuit/a/b$a;
b99bafbbd5288ac93647d22f1c5b1863c96f581ae7a19fdc0e84bff4c2141328.apk.unpack/classes2.dex: 'Lcom/stealien/product/keypadsuit/a/b$b;
b99bafbbd5288ac93647d22f1c5b1863c96f581ae7a19fdc0e84bff4c2141328.apk.unpack/classes2.dex: %Lcom/stealien/product/keypadsuit/b/a;
b99bafbbd5288ac93647d22f1c5b1863c96f581ae7a19fdc0e84bff4c2141328.apk.unpack/classes2.dex: %Lcom/stealien/product/keypadsuit/b/b;
b99bafbbd5288ac93647d22f1c5b1863c96f581ae7a19fdc0e84bff4c2141328.apk.unpack/classes2.dex: %Lcom/stealien/product/keypadsuit/b/c;
b99bafbbd5288ac93647d22f1c5b1863c96f581ae7a19fdc0e84bff4c2141328.apk.unpack/classes2.dex: :Lcom/stealien/product/keypadsuit/button/ClearKeyPadButton;
b99bafbbd5288ac93647d22f1c5b1863c96f581ae7a19fdc0e84bff4c2141328.apk.unpack/classes2.dex: :Lcom/stealien/product/keypadsuit/button/InputKeyPadButton;
b99bafbbd5288ac93647d22f1c5b1863c96f581ae7a19fdc0e84bff4c2141328.apk.unpack/classes2.dex: ;Lcom/stealien/product/keypadsuit/button/NormalKeyPadButton;
b99bafbbd5288ac93647d22f1c5b1863c96f581ae7a19fdc0e84bff4c2141328.apk.unpack/classes2.dex: :Lcom/stealien/product/keypadsuit/encryption/AESEncryption;
b99bafbbd5288ac93647d22f1c5b1863c96f581ae7a19fdc0e84bff4c2141328.apk.unpack/classes2.dex: =Lcom/stealien/product/keypadsuit/encryption/EncryptionMethod;
b99bafbbd5288ac93647d22f1c5b1863c96f581ae7a19fdc0e84bff4c2141328.apk.unpack/classes2.dex: ;Lcom/stealien/product/keypadsuit/generator/KeyPadGenerator;
b99bafbbd5288ac93647d22f1c5b1863c96f581ae7a19fdc0e84bff4c2141328.apk.unpack/classes2.dex: @Lcom/stealien/product/keypadsuit/keypad/listener/OnDrawListener;
b99bafbbd5288ac93647d22f1c5b1863c96f581ae7a19fdc0e84bff4c2141328.apk.unpack/classes2.dex: #Lcom/stealien/product/keypadsuit/R;
b99bafbbd5288ac93647d22f1c5b1863c96f581ae7a19fdc0e84bff4c2141328.apk.unpack/classes2.dex: (Lcom/stealien/product/keypadsuit/R$attr;
b99bafbbd5288ac93647d22f1c5b1863c96f581ae7a19fdc0e84bff4c2141328.apk.unpack/classes2.dex: ,Lcom/stealien/product/keypadsuit/R$drawable;
b99bafbbd5288ac93647d22f1c5b1863c96f581ae7a19fdc0e84bff4c2141328.apk.unpack/classes2.dex: *Lcom/stealien/product/keypadsuit/R$string;
b99bafbbd5288ac93647d22f1c5b1863c96f581ae7a19fdc0e84bff4c2141328.apk.unpack/classes2.dex: )Lcom/stealien/product/keypadsuit/R$style;
b99bafbbd5288ac93647d22f1c5b1863c96f581ae7a19fdc0e84bff4c2141328.apk.unpack/classes2.dex: -Lcom/stealien/product/keypadsuit/R$styleable;
b99bafbbd5288ac93647d22f1c5b1863c96f581ae7a19fdc0e84bff4c2141328.apk.unpack/classes2.dex: -Lcom/stealien/product/keypadsuit/view/KeyPad;
b99bafbbd5288ac93647d22f1c5b1863c96f581ae7a19fdc0e84bff4c2141328.apk.unpack/classes2.dex: /Lcom/stealien/product/keypadsuit/view/KeyPad$1;
b99bafbbd5288ac93647d22f1c5b1863c96f581ae7a19fdc0e84bff4c2141328.apk.unpack/classes2.dex: /Lcom/stealien/product/keypadsuit/view/KeyPad$2;
b99bafbbd5288ac93647d22f1c5b1863c96f581ae7a19fdc0e84bff4c2141328.apk.unpack/classes2.dex: /Lcom/stealien/product/keypadsuit/view/KeyPad$3;
b99bafbbd5288ac93647d22f1c5b1863c96f581ae7a19fdc0e84bff4c2141328.apk.unpack/classes2.dex: /Lcom/stealien/product/keypadsuit/view/KeyPad$4;
b99bafbbd5288ac93647d22f1c5b1863c96f581ae7a19fdc0e84bff4c2141328.apk.unpack/classes2.dex: /Lcom/stealien/product/keypadsuit/view/KeyPad$a;
b99bafbbd5288ac93647d22f1c5b1863c96f581ae7a19fdc0e84bff4c2141328.apk.unpack/classes2.dex: /Lcom/stealien/product/keypadsuit/view/KeyPad$b;
b99bafbbd5288ac93647d22f1c5b1863c96f581ae7a19fdc0e84bff4c2141328.apk.unpack/classes2.dex: /Lcom/stealien/product/keypadsuit/view/KeyPad$c;
b99bafbbd5288ac93647d22f1c5b1863c96f581ae7a19fdc0e84bff4c2141328.apk.unpack/classes2.dex: msg_appsuit_error
b99bafbbd5288ac93647d22f1c5b1863c96f581ae7a19fdc0e84bff4c2141328.apk.unpack/classes2.dex: onKeyPad_AppSuit
b99bafbbd5288ac93647d22f1c5b1863c96f581ae7a19fdc0e84bff4c2141328.apk.unpack/classes2.dex: SEND_LOG_INTERVAL_MIN_APPSUIT
b99bafbbd5288ac93647d22f1c5b1863c96f581ae7a19fdc0e84bff4c2141328.apk.unpack/classes.dex: AppSuit
b99bafbbd5288ac93647d22f1c5b1863c96f581ae7a19fdc0e84bff4c2141328.apk.unpack/classes.dex: APPSUIT
b99bafbbd5288ac93647d22f1c5b1863c96f581ae7a19fdc0e84bff4c2141328.apk.unpack/classes.dex: APP_LOG_APPSUIT
b99bafbbd5288ac93647d22f1c5b1863c96f581ae7a19fdc0e84bff4c2141328.apk.unpack/classes.dex: APP_LOG_APPSUIT_DATE
b99bafbbd5288ac93647d22f1c5b1863c96f581ae7a19fdc0e84bff4c2141328.apk.unpack/classes.dex: getSendLogTimeAppSuit
b99bafbbd5288ac93647d22f1c5b1863c96f581ae7a19fdc0e84bff4c2141328.apk.unpack/classes.dex: Lcom/stealien/const;
b99bafbbd5288ac93647d22f1c5b1863c96f581ae7a19fdc0e84bff4c2141328.apk.unpack/classes.dex: msg_appsuit_error
b99bafbbd5288ac93647d22f1c5b1863c96f581ae7a19fdc0e84bff4c2141328.apk.unpack/classes.dex: PREF_TMONEY_STR_APP_LOG_APPSUIT
b99bafbbd5288ac93647d22f1c5b1863c96f581ae7a19fdc0e84bff4c2141328.apk.unpack/classes.dex: $PREF_TMONEY_STR_APP_LOG_APPSUIT_DATE
b99bafbbd5288ac93647d22f1c5b1863c96f581ae7a19fdc0e84bff4c2141328.apk.unpack/classes.dex: setSendLogTimeAppSuit
b99bafbbd5288ac93647d22f1c5b1863c96f581ae7a19fdc0e84bff4c2141328.apk.unpack/classes.dex: TMONEY_APPSUIT_CHK
b99bafbbd5288ac93647d22f1c5b1863c96f581ae7a19fdc0e84bff4c2141328.apk.unpack/resources.arsc: msg_appsuit_error
e577f62f8646f756b88559296940dbfa53ac3b4a41c831edec75e8d256f35f6d.apk.unpack/classes.dex: AppSuit
e577f62f8646f756b88559296940dbfa53ac3b4a41c831edec75e8d256f35f6d.apk.unpack/classes.dex: APPSUIT
e577f62f8646f756b88559296940dbfa53ac3b4a41c831edec75e8d256f35f6d.apk.unpack/classes.dex: stealien
e577f62f8646f756b88559296940dbfa53ac3b4a41c831edec75e8d256f35f6d.apk.unpack/classes.dex: stealien_aes_key
e577f62f8646f756b88559296940dbfa53ac3b4a41c831edec75e8d256f35f6d.apk.unpack/classes.dex: 1Lcom/stealien/product/keypadsuit/view/KeyPad$2$1;
e577f62f8646f756b88559296940dbfa53ac3b4a41c831edec75e8d256f35f6d.apk.unpack/classes.dex: 5Lcom/stealien/product/keypadsuit/button/KeyPadButton;
e577f62f8646f756b88559296940dbfa53ac3b4a41c831edec75e8d256f35f6d.apk.unpack/classes.dex: 6[Lcom/stealien/product/keypadsuit/button/KeyPadButton;
e577f62f8646f756b88559296940dbfa53ac3b4a41c831edec75e8d256f35f6d.apk.unpack/classes.dex: 8Lcom/stealien/product/keypadsuit/button/DelKeyPadButton;
e577f62f8646f756b88559296940dbfa53ac3b4a41c831edec75e8d256f35f6d.apk.unpack/classes.dex: 8Lcom/stealien/product/keypadsuit/generator/PinGenerator;
e577f62f8646f756b88559296940dbfa53ac3b4a41c831edec75e8d256f35f6d.apk.unpack/classes.dex: 9Lcom/stealien/product/keypadsuit/button/NullKeyPadButton;
e577f62f8646f756b88559296940dbfa53ac3b4a41c831edec75e8d256f35f6d.apk.unpack/classes.dex: BLcom/stealien/product/keypadsuit/generator/ServerSyncPinGenerator;
e577f62f8646f756b88559296940dbfa53ac3b4a41c831edec75e8d256f35f6d.apk.unpack/classes.dex: com.stealien.product.keypadsuit
e577f62f8646f756b88559296940dbfa53ac3b4a41c831edec75e8d256f35f6d.apk.unpack/classes.dex: DLcom/stealien/product/keypadsuit/encryption/DefaultServerEncryption;
e577f62f8646f756b88559296940dbfa53ac3b4a41c831edec75e8d256f35f6d.apk.unpack/classes.dex: DLcom/stealien/product/keypadsuit/keypad/listener/InputEventListener;
e577f62f8646f756b88559296940dbfa53ac3b4a41c831edec75e8d256f35f6d.apk.unpack/classes.dex: getSendLogTimeAppSuit
e577f62f8646f756b88559296940dbfa53ac3b4a41c831edec75e8d256f35f6d.apk.unpack/classes.dex: Lcom/stealien/const;
e577f62f8646f756b88559296940dbfa53ac3b4a41c831edec75e8d256f35f6d.apk.unpack/classes.dex: #Lcom/stealien/product/keypadsuit/a;
e577f62f8646f756b88559296940dbfa53ac3b4a41c831edec75e8d256f35f6d.apk.unpack/classes.dex: %Lcom/stealien/product/keypadsuit/a/a;
e577f62f8646f756b88559296940dbfa53ac3b4a41c831edec75e8d256f35f6d.apk.unpack/classes.dex: %Lcom/stealien/product/keypadsuit/a/b;
e577f62f8646f756b88559296940dbfa53ac3b4a41c831edec75e8d256f35f6d.apk.unpack/classes.dex: 'Lcom/stealien/product/keypadsuit/a/b$a;
e577f62f8646f756b88559296940dbfa53ac3b4a41c831edec75e8d256f35f6d.apk.unpack/classes.dex: 'Lcom/stealien/product/keypadsuit/a/b$b;
e577f62f8646f756b88559296940dbfa53ac3b4a41c831edec75e8d256f35f6d.apk.unpack/classes.dex: %Lcom/stealien/product/keypadsuit/b/a;
e577f62f8646f756b88559296940dbfa53ac3b4a41c831edec75e8d256f35f6d.apk.unpack/classes.dex: %Lcom/stealien/product/keypadsuit/b/b;
e577f62f8646f756b88559296940dbfa53ac3b4a41c831edec75e8d256f35f6d.apk.unpack/classes.dex: %Lcom/stealien/product/keypadsuit/b/c;
e577f62f8646f756b88559296940dbfa53ac3b4a41c831edec75e8d256f35f6d.apk.unpack/classes.dex: :Lcom/stealien/product/keypadsuit/button/ClearKeyPadButton;
e577f62f8646f756b88559296940dbfa53ac3b4a41c831edec75e8d256f35f6d.apk.unpack/classes.dex: :Lcom/stealien/product/keypadsuit/button/InputKeyPadButton;
e577f62f8646f756b88559296940dbfa53ac3b4a41c831edec75e8d256f35f6d.apk.unpack/classes.dex: ;Lcom/stealien/product/keypadsuit/button/NormalKeyPadButton;
e577f62f8646f756b88559296940dbfa53ac3b4a41c831edec75e8d256f35f6d.apk.unpack/classes.dex: :Lcom/stealien/product/keypadsuit/encryption/AESEncryption;
e577f62f8646f756b88559296940dbfa53ac3b4a41c831edec75e8d256f35f6d.apk.unpack/classes.dex: =Lcom/stealien/product/keypadsuit/encryption/EncryptionMethod;
e577f62f8646f756b88559296940dbfa53ac3b4a41c831edec75e8d256f35f6d.apk.unpack/classes.dex: ;Lcom/stealien/product/keypadsuit/generator/KeyPadGenerator;
e577f62f8646f756b88559296940dbfa53ac3b4a41c831edec75e8d256f35f6d.apk.unpack/classes.dex: @Lcom/stealien/product/keypadsuit/keypad/listener/OnDrawListener;
e577f62f8646f756b88559296940dbfa53ac3b4a41c831edec75e8d256f35f6d.apk.unpack/classes.dex: -Lcom/stealien/product/keypadsuit/view/KeyPad;
e577f62f8646f756b88559296940dbfa53ac3b4a41c831edec75e8d256f35f6d.apk.unpack/classes.dex: /Lcom/stealien/product/keypadsuit/view/KeyPad$1;
e577f62f8646f756b88559296940dbfa53ac3b4a41c831edec75e8d256f35f6d.apk.unpack/classes.dex: /Lcom/stealien/product/keypadsuit/view/KeyPad$2;
e577f62f8646f756b88559296940dbfa53ac3b4a41c831edec75e8d256f35f6d.apk.unpack/classes.dex: /Lcom/stealien/product/keypadsuit/view/KeyPad$3;
e577f62f8646f756b88559296940dbfa53ac3b4a41c831edec75e8d256f35f6d.apk.unpack/classes.dex: /Lcom/stealien/product/keypadsuit/view/KeyPad$4;
e577f62f8646f756b88559296940dbfa53ac3b4a41c831edec75e8d256f35f6d.apk.unpack/classes.dex: /Lcom/stealien/product/keypadsuit/view/KeyPad$a;
e577f62f8646f756b88559296940dbfa53ac3b4a41c831edec75e8d256f35f6d.apk.unpack/classes.dex: /Lcom/stealien/product/keypadsuit/view/KeyPad$b;
e577f62f8646f756b88559296940dbfa53ac3b4a41c831edec75e8d256f35f6d.apk.unpack/classes.dex: /Lcom/stealien/product/keypadsuit/view/KeyPad$c;
e577f62f8646f756b88559296940dbfa53ac3b4a41c831edec75e8d256f35f6d.apk.unpack/classes.dex: msg_appsuit_error
e577f62f8646f756b88559296940dbfa53ac3b4a41c831edec75e8d256f35f6d.apk.unpack/classes.dex: SEND_LOG_INTERVAL_MIN_APPSUIT
e577f62f8646f756b88559296940dbfa53ac3b4a41c831edec75e8d256f35f6d.apk.unpack/classes.dex: setSendLogTimeAppSuit
e577f62f8646f756b88559296940dbfa53ac3b4a41c831edec75e8d256f35f6d.apk.unpack/resources.arsc: msg_appsuit_error
e97004b2feb08ee27d85b745b6977f4e7cd6b53fad4eca5c3fb9bdc1655adeb5.apk.unpack/classes.dex: AppSuit
e97004b2feb08ee27d85b745b6977f4e7cd6b53fad4eca5c3fb9bdc1655adeb5.apk.unpack/classes.dex: Lcom/stealien/const;
```

We can match if : `AppSuit` and `stealien` are present. Thus we can set it up to 2 or more. Do you agree?